addons/mass_mailing/models/mailing.py

```diff
@@ -67,6 +67,24 @@ def _get_default_mail_server_id(self):
     email_from = fields.Char(string='Send From', required=True,
         default=lambda self: self.env.user.email_formatted)
     sent_date = fields.Datetime(string='Sent Date', copy=False)
+
+    @api.model
+    def default_get(self, fields):
+        """ When created from the calendar view, we set the schedule_date at the selected date. """
+
+        result = super(MassMailing, self).default_get(fields)
+        default_schedule_date = self.env.context.get('default_schedule_date')
+        if default_schedule_date:
+            result.update({
+                'schedule_type': 'scheduled',
+                'schedule_date': default_schedule_date
+            })
+        else:
+            result.update({'schedule_type': 'now'})
+        return result
+
+    schedule_type = fields.Selection(
+        [('now', 'Send now'), ('scheduled', 'Send on')], string='Schedule')
```

Comment:
Let's set a default to "now" and make it required=True, so the business logic behind can always assume this field has a value.